numpy/core/tests/test_regression.py

```diff
@@ -2375,3 +2375,19 @@ def test_structarray_title(self):
             structure = np.array([1], dtype=[(('x', 'X'), np.object_)])
             structure[0]['x'] = np.array([2])
             gc.collect()
+
+    def test_dtype_scalar_squeeze(self):
+        # gh-11384
+        values = {
+            'S': b"a",
+            'M': "2018-06-20",
+        }
+        for ch in np.typecodes['All']:
+            if ch in 'O':
+                continue
+            sctype = np.dtype(ch).type
+            scvalue = sctype(values.get(ch, 3))
+            for axis in [None, ()]:
```

Comment:
Could add `0` and `-1` here too - surprisingly they're supported

Reply:
I think there's a good case that supporting 0 and -1 on 0d arrays/scalars is a bug.

Reply:
True, but most likely the issue should be deprecated separately.

Reply:
But, still could add it here really. Just also add a comment that we may want to deprecate it (just so that someone who does it is not scared away by a test).